server/src/models/people/models.js

```diff
@@ -1,69 +1,66 @@
-import { camel_case_headcount_model_names } from "./headcount_model_utils.js";
+import _ from "lodash";
+import mongoose from "mongoose";
 
-export default function define_resource_models(model_singleton) {
-  const _headcount_records_by_model_and_dept_code = {};
+import {
+  create_resource_by_id_attr_dataloader,
+  create_resource_by_foreignkey_attr_dataloader,
+} from "../loader_utils.js";
+import {
+  pkey_type,
+  parent_fkey_type,
+  fyear_type,
+  number_type,
+  str_type,
+  bilingual,
+} from "../model_utils.js";
 
-  class HeadcountModel {
-    constructor(obj) {
-      Object.assign(this, obj);
-    }
-    static model_name() {
-      throw new Error(`
-        Error: bad OOP alert, don't use HeadcountModels class directly.
-        Need to create a new class extending it that overwrites the static 
-        model_name() function with one that returns the name of the actual model.
-      `);
-    }
-    static get_model_name() {
-      try {
-        return this.model_name();
-      } catch (e) {
-        console.log(e);
-      }
-    }
-    static register(obj) {
-      const { dept_code } = obj;
+export default function (model_singleton) {
+  const EmployeeWholeDataSchema = mongoose.Schema({
```

Comment:
Small name nitpick, because "whole data" doesn't mean too much to me, I'd call these `EmployeePopulationDataSchema` or something. The general family of datasets it applies to is the "population" data (different ways of slicing up the same total population numbers, all those datasets sum to the same total).